system/lib/wasmfs/js_api.cpp

```diff
@@ -264,6 +266,39 @@ int _wasmfs_lstat(char* path, struct stat* statBuf) {
   return __syscall_lstat64((intptr_t)path, (intptr_t)statBuf);
 }
 
+int _wasmfs_mount(char* path, backend_t created_backend) {
+  int err = __syscall_rmdir((intptr_t)path);
+  assert(!err);
+
+  if (err == -ENOTEMPTY) {
+    // Check for an attempt to mount to an existing mountpoint.
```

Comment:
I saw in `test_mount.c` that there was a specific test where `-EBUSY` was returned if the mountpoint already exists, so I thought it might be helpful to keep compatibility with the legacy JS API. Would returning `-ENOTEMPTY` be ideal here instead of trying to determine if `-EBUSY` is needed?

Reply:
Yeah, I think just returning `ENOTEMPTY` would be good enough, even if it's different than what the legacy API does. If this required less code to match the legacy behavior, then it might have been worth it.